python/pip_install/pip_compile.py

```diff
@@ -14,7 +14,12 @@
     sys.exit(1)
 
 requirements_in = os.path.relpath(sys.argv.pop(1))
-requirements_txt = sys.argv.pop(1)
+requirements_txt = os.path.relpath(sys.argv.pop(1))
+parts = requirements_in.split(os.path.sep, 2)
+if parts[0] == "external":
+    requirements_in = parts[2]
```

Comment:
shouldn't this be all the segments joined other than the first two? what if requirements.in isn't in the root of the external repository, but is in `external/other_repo/some_folder/requirements.in` ?

Reply:
In this case, `parts[2]` would be `some_folder/requirements.in`. Then it should work as expected.

Reply:
ah, I see the `split` is limited to two